src/software/ai/navigator/path_planner/theta_star_path_planner_test.cpp

```diff
@@ -72,22 +68,17 @@ TEST(TestThetaStarPathPlanner, test_theta_star_path_planner_blocked_src)
         Polygon({Point(0.5, 1), Point(-0.5, 1), Point(-0.5, -1), Point(0.5, -1)}))};
 
     std::unique_ptr<PathPlanner> planner = std::make_unique<ThetaStarPathPlanner>();
-    std::vector<Point> path_points;
 
-    try
-    {
-        path_points = std::get<std::vector<Point>>(
-            planner->findPath(start, dest, field, obstacles));
-    }
-    catch (const std::bad_variant_access&)
-    {
-        // not a valid path
-        FAIL();
-    }
+    Rectangle navigableArea(Point(field.totalXLength(), field.totalYLength()),
+                            field.totalXLength(), field.totalYLength());
+
+    Path path = planner->findPath(start, dest, navigableArea, obstacles);
 
     // Make sure the start and end of the path are correct
-    EXPECT_EQ(start, path_points.front());
-    EXPECT_EQ(dest, path_points.back());
+    EXPECT_EQ(start, path.calculateValue(0));
+    EXPECT_EQ(dest, path.calculateValue(path.size()));
```

Comment:
Previously we used `1` for the dest, now we're using `path.size()`? I'm quite confused.

Reply:
path.size is the end of the path: Spline takes in points and assigns each point to a unit of time (e.g. time=0 -> point 1, time=1 -> point 2, ...) and interpolates with line segments. Line segment 1 goes from time=0 to time=1 and from point 1 to point 2, etc.

Reply:
Ah, I see. In the c++ std library, `size` generally refers to the number of elements, hence my confusion. However I also don't think that we should use the term `length` here, because that generally refers to arc length, which this absolutely is not. I think the easiest option is just to add a `Path::end()` function to get the final point.

Reply:
See new changes, this might be addressed

Reply:
I still think it would make things more readable if we had `startPoint` and `endPoint` methods

Reply:
done